RELEASES.md

```diff
@@ -8,6 +8,11 @@ Active development is happening on the `main` branch, and a new version is relea
 
 Stable releases of Envoy include:
 
+TODO(RyanTheOptimist): It seems to me that we have two kinds of releases and it would be helpful for
+this doc to make a clear distinction between them. For the sake of argument, I've called them
+"Major Release" (v1.22) and "Security Releases" (v1.22.1, etc). Does this terminology work?
```

Comment:
I think we effectively have 3 (well 2 with 1 having 2 subtypes):
1) Major release
2) Minor release (security)
3) Minor release (stability)

The release mechanics for 2 and 3 are very similar, there is just more private coordination required for 2.

Reply:
They are basically ad-hoc and best effort. It's essentially if someone wants to do the work. I agree we should discuss this option and make it clear that it's best effort and requires people to volunteer for release work.